RadixWallet/Features/TransactionReviewFeature/TransactionReview+Sections.swift

```diff
@@ -344,6 +350,23 @@ extension TransactionReview {
 		return DappEntity(id: dAppDefinitionAddress, metadata: metadata)
 	}
 
+	private func extractDappEntitiesFromLockerAddresses(_ addresses: [LockerAddress]) async -> [(address: Address, entity: DappEntity?)] {
+		await addresses.asyncMap {
+			await (address: $0.asGeneral, entity: try? extractDappEntityFromLockerAddress($0))
+		}
+	}
+
+	private func extractDappEntityFromLockerAddress(_ address: LockerAddress) async throws -> DappEntity? {
+		guard
+			let locker = try await onLedgerEntitiesClient.getEntity(address.asGeneral, metadataKeys: .dappMetadataKeys).locker,
+			let dAppDefinitionAddress = locker.metadata.dappDefinition
+		else {
+			return nil
+		}
+		let metadata = try await onLedgerEntitiesClient.getDappMetadata(dAppDefinitionAddress)
+		return DappEntity(id: dAppDefinitionAddress, metadata: metadata)
+	}
```

Comment:
hmm, is extracting DappEntity from Locker is different compared to other components? I would expect the current `extractDappEntity` to be used for lockers as well.

Reply:
updated